### DISCUSSION TITLE
# 43 : sometimes its really annoying to get 100% 😑😑

### DISCUSSION THREAD

**Discussion by thoughtlessnerd** [cssbattle/forum#52] [General]
![image](https://user-images.githubusercontent.com/79973672/114707356-bfbcea80-9d47-11eb-9526-d0f4fadd4d22.png)

here's the code i used : 
`<p><style>*{background:#6592CF;position:fixed}*>*{width:57;height:182;border-radius:0+70px+70px+0;left:90;top:22;border:30px solid#243D83;-webkit-box-reflect:right -30px}p{width:402;height:301;left:-30;top:-48;border-radius:0;background:linear-gradient(90deg,#6592CF 145px,transparent 0`

i even used the server side previewer but nothing i could find out

here's the difference between my code render on server and the final result. 😶

![image](https://user-images.githubusercontent.com/79973672/114707741-34902480-9d48-11eb-8a4a-714ee1d99994.png)

PLEASE HELP !

**Comment by M4N0JKUM4R** [comment]
I have never used a property like [`-webkit-box-reflect`](https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-box-reflect)  as it is non-standard. This is how it is rendered in Firefox: 

![image](https://user-images.githubusercontent.com/11471878/114821189-d1e96800-9ddd-11eb-9fca-4643f056353b.png) 

I would just stick to a normal method that I used with same character count(-2):

`<div a><div b></div><div c></div></div><style>body{background:#6592CF;}[a]{width:110;height: 242;transform:translate(137px,22px);overflow:hidden}[b],[c]{width:100;height:182;border:30px solid #243D83;border-radius:70px;transform:translate(40px,0)}[c]{transform:translate(-90px,-242px)`

Yes, please replace `div` with your preferred element and adjust the positioning.

**Reply by thoughtlessnerd** [reply]
server uses chromium or chrome as its renderer so its `-webkit-box-reflect` fine to use sometimes where `box-shadow` can't do the job. by the way a tip: use [https://codepen.io/alexzaworski/embed/vwmNKr](server side render previewer) to do these challenges. its pretty useful for unicode based solutions too.
like f you are wondering how #69 is 100% still there is just a box on the result then go and paste the code on this previewer and you will know.

**Reply by Beowolve** [reply]
-webkit-box-reflect is no problem with this target, I use it and reached 143 chars with it

**Comment by M4N0JKUM4R** [comment]
I have a similar situation on this one:

`<div i></div><div j></div><div k></div><div l></div><div m></div><div n></div><div o></div><style> body{background:#1A4341}div{width:300;height:20;background:#F3AC3C;margin-bottom:20;transform: translate(50px,52px)} div[n],div[O]{width:300;height:300;background:#1A4341;border-radius:50%; transform:translate(-158px,-208px)}div[O]{transform:translate(242px,-528px)`

![image](https://user-images.githubusercontent.com/11471878/114822582-e9c1eb80-9ddf-11eb-842e-8c833fb44ba4.png)

Looks error free to me though :P

**Reply by iamvdo** [reply]
For this one (and many others), what matters is where colors meet at sharp edges, eg at corners. I tweak your code a bit to fix corners (remove 2px from height and adjust transforms), and even if it does not match elsewhere, it gives 100%, see:

`<div i></div><div j></div><div k></div><div l></div><div m></div><div n></div><div o></div><style> body{background:#1A4341}div{width:300;height:20;background:#F3AC3C;margin-bottom:20;transform: translate(50px,52px)} div[n],div[O]{width:300;height:298;background:#1A4341;border-radius:50%; transform:translate(-157px,-207px)}div[O]{transform:translate(241px,-525px)`

**Reply by M4N0JKUM4R** [reply]
I fail to understand the scoring system. I tried your code, yes it matches 100% but from a viewer's point would be below 99. 


https://user-images.githubusercontent.com/11471878/114876161-ae451280-9e1b-11eb-8931-6c963ad76fb8.mp4

**Reply by iamvdo** [reply]
It's all about antialiasing. Drawing shapes in a graphical editor is not the same as using CSS from a browser (authors said they don't use CSS to create target's visuals), but even in a browser you won't get exact match between browsers and systems (not same rounding values, compositing strategies, etc.). 🤷‍♂️

**Reply by iamvdo** [reply]
Well, in the same time, it doesn't explain why those corners need to be "almost" perfect while other parts doesn't.

**Comment by iamvdo** [comment]
@JavaScriptKing007  The shape is looking good, but you have a 1px line at the bottom. Use `p{height:302}` instead of `p{height:301}`

**Reply by thoughtlessnerd** [reply]
i tried, that doesn't change anything. i guess i will try to do in some longer way to atleast get 600 and then start optimizing.

**Reply by iamvdo** [reply]
But it gives 100%. Isn't it what you're trying to achieve? 😅

`<p><style>*{background:#6592CF;position:fixed}*>*{width:57;height:182;border-radius:0+70px+70px+0;left:90;top:22;border:30px solid#243D83;-webkit-box-reflect:right -30px}p{width:402;height:302;left:-30;top:-48;border-radius:0;background:linear-gradient(90deg,#6592CF 145px,transparent 0`

**Reply by thoughtlessnerd** [reply]
hell yeah it is, thanks man.

**Comment by thoughtlessnerd** [comment]
By the way @iamvdo , How do you do these challenges in just 150 chars ?? like is there some type of trick or hack ?
![image](https://user-images.githubusercontent.com/79973672/114970294-cce8ef00-9e97-11eb-8d1e-b6c51ee5bef5.png)

**Reply by M4N0JKUM4R** [reply]
Well, I though it could be one of the following: https://github.com/cssbattle/forum/discussions/34#discussioncomment-608232 looking at his solutions #69 but that looks out of case: 

    ᐢ ǁ ᐢ<img>◆<img><style>img{background:radial-gradient(#0000 48q,#E08027 0);margin:-89-49 13;border-radius:var(--,0)50%50%}img+*{--:50%0}*{padding:57;margin:11-3;color:E08027;border:solid 10px#191919;background:#191919;font:700 76q/0'

This is how it is rendered in Chrome but server side render is better: 

![image](https://user-images.githubusercontent.com/11471878/114974964-03773780-9ea1-11eb-86f7-97c03e606d12.png)

It comes down to understanding of the power of [radial-gradient](https://developer.mozilla.org/en-US/docs/Web/CSS/radial-gradient()).

**Reply by iamvdo** [reply]
Well, there is no hack. "Open targets" are a bit misleading because only two are not based on unicode ([#71](https://cssbattle.dev/play/71)) and ([#73](https://cssbattle.dev/play/73)), but on all 76 targets, maybe only 10-15 are unicode based. All others are "just" CSS. 

[Target #43](https://cssbattle.dev/play/43) is one of them. Only HTML elements with borders, rounded corners, etc.

Another quick tip: I've never used `position:absolute`, `fixed` or whatever. Also, never used `transform` for translations only. ;)

**Reply by thoughtlessnerd** [reply]
Then what you use for positioning if you don't use `position:absolute || fixed`??
and about `transform`, i realized it pretty early that its useless to use `transform` for only translations 😅

**Reply by iamvdo** [reply]
Well, mostly `margin` and `padding`.

**Reply by asieduernest12** [reply]
i tried the padding approach. it works till i have to rotate something because i just nest my elements
